libr/util/annotated_code.c

```diff
@@ -9,10 +9,28 @@ R_API RAnnotatedCode *r_annotated_code_new(char *code) {
 		return NULL;
 	}
 	r->code = code;
-	r_vector_init (&r->annotations, sizeof (RCodeAnnotation), NULL, NULL);
+	r_vector_init (&r->annotations, sizeof (RCodeAnnotation), r_annotation_free, NULL);
 	return r;
 }
 
+R_API void r_annotation_free(void *e, void *user) {
+	(void)user;
+	RCodeAnnotation *annotation = e;
+	if (annotation->type == R_CODE_ANNOTATION_TYPE_FUNCTION_NAME) {
+		free (annotation->reference.name);
+	} else if (annotation->type == R_CODE_ANNOTATION_TYPE_LOCAL_VARIABLE || annotation->type == R_CODE_ANNOTATION_TYPE_FUNCTION_PARAMETER) {
+		free (annotation->variable.name);
+	}
+}
+
+R_API bool r_annotation_is_reference(RCodeAnnotation *annotation) {
+	return (annotation->type == R_CODE_ANNOTATION_TYPE_GLOBAL_VARIABLE || annotation->type == R_CODE_ANNOTATION_TYPE_CONSTANT_VARIABLE || annotation->type == R_CODE_ANNOTATION_TYPE_FUNCTION_NAME);
```

Comment:
Can you make this LOC shorter by splitting the conditions in new lines? Had to enable line-wrap here

Reply:
@ITAYC0HEN Can you tell me how I can do that? When I tried it earlier, the coding style checker was telling me to not split it into multiple lines.

Reply:
Then I have no idea :)
That's only a small nice-to-have, if it causes a problem then no worries, you can skip this request